`BlockingRawTest` is flaky

### Description

E.g.:
- https://github.com/quarkusio/quarkus/pull/23808#issuecomment-1045183815
- https://github.com/quarkusio/quarkus/pull/23763#issuecomment-1042891973
- https://github.com/quarkusio/quarkus/pull/23648#issuecomment-1037962015

```
2022-02-18T20:29:35.0348247Z [INFO] Running com.example.grpc.hibernate.BlockingRawTest
2022-02-18T20:30:05.9362427Z no of elements: 100
2022-02-18T20:30:05.9394383Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 30.896 s <<< FAILURE! - in com.example.grpc.hibernate.BlockingRawTest
2022-02-18T20:30:05.9395346Z [ERROR] com.example.grpc.hibernate.BlockingRawTest.shouldAdd  Time elapsed: 30.459 s  <<< ERROR!
2022-02-18T20:30:05.9396516Z org.awaitility.core.ConditionTimeoutException: Condition with lambda expression in com.example.grpc.hibernate.BlockingRawTest that uses java.util.List was not fulfilled within 30 seconds.
2022-02-18T20:30:05.9399484Z 	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:164)
2022-02-18T20:30:05.9449910Z 	at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
2022-02-18T20:30:05.9450474Z 	at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
2022-02-18T20:30:05.9450980Z 	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:939)
2022-02-18T20:30:05.9451482Z 	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:908)
2022-02-18T20:30:05.9452368Z 	at com.example.grpc.hibernate.BlockingRawTest.shouldAdd(BlockingRawTest.java:60)
```
(notice the delay of 30s)

### Implementation ideas

_No response_

--- a/integration-tests/grpc-hibernate/src/test/java/com/example/grpc/hibernate/BlockingRawTest.java
+++ b/integration-tests/grpc-hibernate/src/test/java/com/example/grpc/hibernate/BlockingRawTest.java
@@ -6,6 +6,7 @@ import static org.awaitility.Awaitility.await;
 import java.time.Duration;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Timeout;
@@ -49,7 +50,7 @@ public class BlockingRawTest {
                     .await().atMost(Duration.ofSeconds(5));
         }
 
-        List<String> actual = new ArrayList<>();
+        List<String> actual = new CopyOnWriteArrayList<>();
         Multi<TestOuterClass.Item> all = client.getAll(EMPTY)
                 .onFailure().invoke(th -> {
                     System.out.println("Failed to read");
@@ -65,8 +66,8 @@ public class BlockingRawTest {
     @Timeout(TIMEOUT)
     void shouldAddViaBidi() {
         List<String> expected = new ArrayList<>();
-        List<String> echoed = new ArrayList<>();
-        List<String> actual = new ArrayList<>();
+        List<String> echoed = new CopyOnWriteArrayList<>();
+        List<String> actual = new CopyOnWriteArrayList<>();
 
         Multi<TestOuterClass.Item> request = Multi.createFrom().emitter(
                 m -> {

